src/mod/l4_thruster.c

```diff
@@ -541,27 +741,1590 @@ void GeneralReadRequest(int argc, char *argv[]){
 }
 
 
-//
-//// This will be moved to ado lib later. see https://github.com/RobertK66/ado-chip-175x-6x/issues/25
-///* Update CRC8 Checksum */
-//void c_CRC8(char data, uint8_t *checksum)
-//{
-//    uint8_t i;
-//    *checksum ^= data;
-//
-//    for (i = 0; i < 8; ++i)
-//    {
-//        *checksum = (*checksum << 1) ^ ((*checksum & 0x80) ? 0x07 : 0x00);
-//    }
-//}
-//
-///* Compute CRC8 (binary String) */
-//uint8_t CRC8_thruster(uint8_t* str, size_t length)
-//{
-//    uint8_t checksum = 0;
-//
-//    for (; length--; c_CRC8(*str++, &checksum))
-//        ;
-//
-//    return checksum;
-//}
+
+
+//////
+void thr_wait(int argc, char *argv[]){
+	LAST_STARTED_MODULE = 1108;
+
+	/*
+	 * thr_wait is a "staging" function incorporated in THR_EXECUTION_SEQUENCE
+	 * at end of previous stage - timestep of stage completion is recorded
+	 *
+	 * duration is required input from argv
+	 *
+	 * function will do nothing untill difference between current and previous timestamps is less then duration
+	 *
+	 * THR_EXECUTION_INDEX points to next action in the execution sequence stack.
+	 * THR_EXECUTION INDEX is increased after current timestamp increased above designed wait duration
+	 *
+	 */
+	uint16_t procedure_id = atoi(argv[0]); // procedure_id is always fist index of argument array
+	uint32_t duration = atoi(argv[1]);
+	uint32_t now_timestamp = (uint32_t)timGetSystime();
+
+
+	char print_str[200];
+	int len;
+
+	if ( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage) < duration ){
+		// do nothing
+	}
+	else{
+		THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage = (uint32_t)timGetSystime(); //save execution finish time of wait stage
+		sprintf(print_str, "\nWait Stage= %d Complete t = %d\n", THR_HARDCODED_SEQUENCES[procedure_id].execution_index,now_timestamp);
+		len = strlen(print_str);
+		deb_print_pure_debug((uint8_t *)print_str, len);
+		THR_HARDCODED_SEQUENCES[procedure_id].execution_index++; // increase sequence execution index so that after wait - next module to be executed
+
+		//*******assume that argv[5] is custom print message !!!!! WARNING I AM NOT SURE THAT THIS IS GOOD IDEA
+		len = strlen(argv[5]);
+		deb_print_pure_debug((uint8_t *)argv[5], len);
+		//// WARNING THIS BLOCK MIGHT BE NO GOOD *********
+	}
+
+}
+
+
+void thr_wait_and_monitor(int argc, char *argv[]){
+	/*
+	 *
+	 * This module is use for Hot Standby Script Sequence
```

Comment:
Not only for Hot Standby 

For all sequences that require to wait and monitor. 

I will need to update this function WHEN CLIENT (ENPULSION) or CLIMB system engineers will provide more details on what have to be monitored (and what to do when things go wrong, when to interrupt sequence)